cookbooks/bcpc-hadoop/recipes/resource_manager.rb

```diff
@@ -54,6 +54,9 @@
   returns [0, 1]
 end
 
+hdfs_write = "echo 'test' | hdfs dfs -copyFromLocal - /user/hdfs/chef-mapred-test"
+hdfs_remove = "hdfs dfs -rm -skipTrash /user/hdfs/chef-mapred-test"
```

Comment:
These two lines are just declaring a variable to store the commands. It doesn't execute these command. The variables are being used on line 70 that was getting too long with these two commands. I was able to upload the cookbook and chef the node without any compilation errors:
```
vagrant@bcpc-bootstrap:~/chef-bcpc$ git diff cookbooks/bcpc-hadoop/recipes/resource_manager.rb
diff --git a/cookbooks/bcpc-hadoop/recipes/resource_manager.rb b/cookbooks/bcpc-hadoop/recipes/resource_manager.rb
index a779b47..cbd4bcb 100644
--- a/cookbooks/bcpc-hadoop/recipes/resource_manager.rb
+++ b/cookbooks/bcpc-hadoop/recipes/resource_manager.rb
@@ -54,6 +54,9 @@ bash "kill yarn-resourcemanager" do
   returns [0, 1]
 end

+hdfs_write = "echo 'test' | hdfs dfs -copyFromLocal - /user/hdfs/chef-mapred-test"
+hdfs_remove = "hdfs dfs -rm -skipTrash /user/hdfs/chef-mapred-test"
+
 bash "setup-mapreduce-app" do
   code <<-EOH
   hdfs dfs -mkdir -p /hdp/apps/#{node[:bcpc][:hadoop][:distribution][:release]}/mapreduce/
@@ -64,7 +67,7 @@ bash "setup-mapreduce-app" do
   EOH
   user "hdfs"
   not_if "hdfs dfs -test -f /hdp/apps/#{node[:bcpc][:hadoop][:distribution][:release]}/mapreduce/mapreduce.tar.gz", :user => "hdfs"
-  only_if "echo 'test' | hdfs dfs -copyFromLocal - /user/hdfs/chef-mapred-test", :user => "hdfs"
+  only_if "#{hdfs_write} && #{hdfs_remove}", :user => "hdfs"
 end

 bash "delete-temp-file" do
vagrant@bcpc-bootstrap:~/chef-bcpc$ knife cookbook upload bcpc-hadoop
[/opt/chef/embedded/lib/ruby/gems/1.9.1/specifications/rake-compiler-0.9.5.gemspec] isn't a Gem::Specification (NilClass instead).
Uploading bcpc-hadoop    [0.1.0]
Uploaded 1 cookbook.

During chef-run:
  * template[/etc/init.d/hadoop-yarn-resourcemanager] action create (up to date)
  * bash[setup-mapreduce-app] action run (skipped due to not_if)
  * bash[delete-temp-file] action nothing (skipped due to action :nothing)

```
Did you get a compile time error with this code?